src/litserve/specs/openai_embedding.py

```diff
@@ -0,0 +1,99 @@
+# Copyright The Lightning AI team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import asyncio
+import inspect
+import logging
+import time
+import uuid
+from typing import List, Literal, Optional, Union
+
+from fastapi import Request, Response, status
+from pydantic import BaseModel
+
+from litserve.specs.base import LitSpec
+from litserve.utils import LitAPIStatus
+
+logger = logging.getLogger(__name__)
+
+
+class EmbeddingRequest(BaseModel):
+    input: Union[str, List[str]]
+    model: str
+    dimensions: Optional[int] = None
+    encoding_format: Literal["float"] = "float"
+
+
+class Embedding(BaseModel):
+    index: int
+    embedding: List[float]
+    object: Literal["embedding"] = "embedding"
+
+
+class UsageInfo(BaseModel):
+    prompt_tokens: int = 0
+    total_tokens: int = 0
+
+
+class EmbeddingResponse(BaseModel):
+    data: List[Embedding]
+    model: str
+    object: Literal["list"] = "list"
+    usage: UsageInfo
+
+
+class OpenAIEmbeddingSpec(LitSpec):
```

Comment:
probably we should inherit `OpenAISpec` and you can avoid some repetition in `setup` method.

Reply:
The code is short enough that I'd keep it flat.

Rather in the future we could have a `OpenAISpec` class and then specialized spec classes `OpenAIEmbeddingSpec` and `OpenAIChatCompletionSpec`.

We could then configure `OpenAISpec(api="chat-completion")` (default, this way current users will not be impacted) and `OpenAISpec(api="embedding")`, so the user would only ever see `OpenAISpec` and configure it with a string.

Not strictly needed right now and we can merge as is, but what's your take on this?

Reply:
Coincidentally, I was also thinking of something similar, breaking `ChatCompletionSpec` out from `OpenAISpec` as Luca has suggested. In the future, if we plan - it could be extended to include TTS, etc.

Reply:
yeah, let's not start nesting things like @lantiga suggested. We want to keep things as flat as possible and it is okay to have duplication (cc @aniketmaurya )